openrouteservice/src/main/java/org/heigit/ors/matrix/algorithms/dijkstra/DijkstraManyToMany.java

```diff
@@ -0,0 +1,565 @@
+        ListIterator<AveragedMultiTreeSPEntry> it = entries.listIterator();
+        while (it.hasNext()) {
+            AveragedMultiTreeSPEntry listEntry = it.next();
+            if (listEntry.getEdge() == iter.getEdge()) {
+                entry = listEntry;
+                break;
+            }
+        }
+        return entry;
+    }
+
+    /**
+     * Generate the list of entries for a given node. Initialize the target node in the normal weight map if none exists
+     * @param iter Iterator with adj node to initialize
+     * @return list of entries
+     */
+    private List<AveragedMultiTreeSPEntry> createEntriesList(EdgeIterator iter) {
+        List<AveragedMultiTreeSPEntry> entries;
+        entries = initBestWeightMapEntryList(bestWeightMapCore, iter.getAdjNode());
+        //Initialize target entry in normal weight map
+        if(coreExitPoints.contains(iter.getAdjNode())){
+            AveragedMultiTreeSPEntry target = bestWeightMap.get(iter.getAdjNode());
+            if (target == null) {
+                target = createEmptyEntry(iter);
+                bestWeightMap.put(iter.getAdjNode(), target);
+            }
+        }
+        return entries;
+    }
+
+    List<AveragedMultiTreeSPEntry> initBestWeightMapEntryList(IntObjectMap<List<AveragedMultiTreeSPEntry>> map, int traversalId) {
+        if (map.get(traversalId) != null)
+            throw new IllegalStateException("Core entry point already exists in best weight map.");
+
+        List<AveragedMultiTreeSPEntry> entryList = new ArrayList<>(5);
+        map.put(traversalId, entryList);
+
+        return entryList;
+    }
+
+    /**
+     *
+     * @return whether all goal nodes have been found
+     */
+    private boolean finishedDownwards() {
+        //First check whether all targets found for all sources
+        return stoppingCriterion.isFinished(currEdge, prioQueue);
+
+    }
+
+    private void configureTurnWeighting(EdgeIteratorState iter, MultiTreeSPEntryItem currEdgeItem) {
+        if(hasTurnWeighting && !isInORS(iter, currEdgeItem))
+            turnWeighting.setInORS(false);
+    }
+
+    private void resetTurnWeighting() {
+        if(hasTurnWeighting)
+            turnWeighting.setInORS(true);
+    }
+
+    /**
+     * Check whether the turnWeighting should be in the inORS mode. If one of the edges is a virtual one, we need the original edge to get the turn restriction.
+     * If the two edges are actually virtual edges on the same original edge, we want to disable inORS mode so that they are not regarded as u turn,
+     * because the same edge id left and right of a virtual node results in a u turn
+     * @param iter
+     * @param currEdgeItem
+     * @return
+     */
+    private boolean isInORS(EdgeIteratorState iter, MultiTreeSPEntryItem currEdgeItem) {
+        return currEdgeItem.getEdge() == iter.getEdge()
+                || currEdgeItem.getOriginalEdge() != EdgeIteratorStateHelper.getOriginalEdge(iter);
+    }
```

Comment:
As far as I can tell the code from `configureTurnWeighting` and `resetTurnWeighting` are common parts used across both `DijkstraManyToMany` and `CoreMatrixAlgorithm`. As such, don't you think they could be abstracted together with d`isInORS` into static helper functions living in a separate class?

Reply:
:+1: done

Reply:
Great, thanks a lot for making the effort of refactoring this!